mlflow/dspy/load.py

```diff
@@ -41,6 +42,11 @@ def _load_model(model_uri, dst_path=None):
 
     _set_dependency_schema_to_tracer(local_model_path, loaded_wrapper.dspy_settings["callbacks"])
 
+    mlflow_model = Model.load(local_model_path)
+    if mlflow_model.model_id:
+        # the model of the wrapper will be used for inference
+        _MODEL_TRACKER.set(loaded_wrapper.model, mlflow_model.model_id)
+
```

Comment:
Pyfunc already has its own model tracking logic. Can we check if it doesn't conflict `_MODEL_TRACKER`?

Reply:
Ah didn't realize that, let me add a test case :)